lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -307,6 +307,89 @@ def test_index_no_visible_section_in_chapter(self):
         self.assertNotIn('Problem 1', response.content)
         self.assertNotIn('Problem 2', response.content)
 
+    def _create_staff_user_and_course(self):
+        """
+        User will have staff access and creates the course
+        """
+        self.user = GlobalStaffFactory.create()
+        self.client.login(username=self.user.username, password='test')
+
+        # create the course
+        course = CourseFactory.create()
+        return course
+
+    def _create_url_for_enroll_staff(self, course):
+        """
+        creates the courseware url and enroll staff url
+        """
+        # create the _next parameter
+        courseware_url = reverse('courseware', kwargs={
+            'course_id': unicode(course.id)}) + '?activate_block_id=test_block_id'
+
+        # create the url for enroll_staff view
+        enroll_staff_url = "{enroll_staff_url}?next={courseware_url}".format(
+            enroll_staff_url=reverse('enroll_staff', kwargs={'course_id': unicode(course.id)}),
+            courseware_url=courseware_url
+        )
+        return enroll_staff_url
+
+    def test_redirection_unenrolled_staff(self):
+        """
+        Test that staff is not redirected to the 'about' page when visiting an unregistered course
+        """
+        course = self._create_staff_user_and_course()
+        enroll_staff_url = self._create_url_for_enroll_staff(course)
+
+        # Check the GET method
+        response = self.client.get(enroll_staff_url)
```

Comment:
The docstring for the tests says that you're checking to make sure that staff is redirected to the enroll page — I think this will be covered by `test_courseware_redirection` below, so this test doesn't seem necessary